File: Natro-Backend/Natro-Backend.Entity/Entities/UserFavoriteEntity.cs
using Natro_Backend.Entity.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Natro_Backend.Entity.Entities
{
    public class UserFavoriteEntity : BaseEntity
    {
        public string Domain { get; set; }
        public bool IsAvailableToBuy { get; set; }
        public string Nameserver1 { get; set; }
        public string Nameserver2 { get; set; }
        public DateTime LastChange { get; set; }
        public DateTime ExpireDate { get; set; }
        public int UserID { get; set; }

        [ForeignKey("UserID")]
        public virtual UserEntity User { get; set; }
        public virtual List<UserFavoriteNotificationEntity> UserFavoriteNotifications { get; set; }
    }
}

File: natro-ui/src/app/core/models/entities/userFavoriteEntity.ts
import { BaseEntity } from "./baseEntity";
import { UserEntity } from "./userEntity";

export class UserFavoriteEntity extends BaseEntity {
    domain: string;
    isAvailableToBuy: boolean;
    expireDate: Date;
    lastChange: Date;
    nameserver1: string;
    nameserver2: string;
    userID: number;
    user: UserEntity;
}